docs/prover/diagnosis/index.md

```diff
@@ -5,15 +5,250 @@ Diagnostic Tools
 (tac-reports)=
 ## TAC Reports
 
-% TODO  write this -- https://certora.atlassian.net/browse/DOC-355
+{term}`TAC` Reports provide an under-the-hood view on a given
+{term}`verification condition` as well as the result that the Prover produced
+for that verification condition, if available. There are four variants
+of TAC reports, one each for the results SAT, UNSAT, TIMEOUT, and one that
+contains no information from the result. In the following, we will discuss these
+variants one by one. We will begin with the TAC report without Prover result
+information, since its constituents are present in the other variants as well.
 
-% ## Timeout TAC Reports
-% 
-% ### Statistics- and Explanation-Box
-% 
-% ### Split- and Heuristical Difficulty-Coloring
-% 
-% ### TAC Source Code Box
-
-There is a brief explanation of how to use TAC reports in the 
+```{note}
+In addition to the information below, there is a brief explanation of how to 
+use TAC reports in the 
 [webinar on timeouts](https://www.youtube.com/watch?v=mntP0_EN-ZQ).
+```
+
+### Plain TAC reports
+
+
+```{figure} rebase-tac-report-plain-annotated.png
+:name: plain-tac-report
+:alt: Plain TAC report example
+:align: center
+
+Example of a plain TAC report (with annotations in red)
+```
+
+At the center of a TAC report is a visualization of the verification condition's
+{term}`control flow graph` (CFG).[^nested-cfg] There are two kinds of nodes.
+Regular nodes and call nodes. Regular nodes have a solid outline, while call
+nodes have a dashed outline. Clicking on a regular nodes will make the source
+code box (discussed below) focus on the corresponding TAC source code; clicking
+on a call node will replace the currently displayed CFG with the CFG that
+belongs to the called method. 
+
+```{note}
+Only external calls are explicit in the TAC report's CFGs. Internal calls are 
+inlined on the TAC source code level.
+```
+
+[^nested-cfg]: Strictly speaking, there is a set of CFGs available for each
+    verification condition. Every external call has its own CFG, and the CFGs
+    are related by call nodes which lead from a call site to the corresponding
+    callee's CFG. Intuitively, this set can be viewed as one CFG with nested 
+    sub-CFGs for the calls.
+
+The upper-mid left part of a TAC report contains the TAC source code. The
+details of the TAC language are outside the scope of this documentation.
+Generally speaking, TAC is generated by compiler-style transformations of the
+EVM bytecode together with the CVL specification. Many patterns from EVM
```

Comment:
`EVM bytecode` - we generate TAC dumps also for Solana and WASM inputs. Though they are still not much supported yet (mainly WASM is not much supported). So perhaps keep the EVM bytecode there for now, but keep in mind it would be good to update it a few months later